src/Symfony/Component/Mime/Email.php

```diff
@@ -485,6 +485,8 @@ private function prepareParts(): ?array
                 $attachment['inline'] = true;
                 $inlineParts[$name] = $part = $this->createDataPart($attachment);
                 $html = str_replace('cid:'.$name, 'cid:'.$part->getContentId(), $html);
+                $part->setName($part->getContentId());
+                $attachment['name'] = $part->getContentId();
```

Comment:
this line is useless as `$attachment` is never read again after that assignment

Reply:
you're right. I missed the ```continue 2``` that would break the outer loop as well.